LakeExport.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+@author: simontopp
+"""
+#%%
+import time
+import ee
+import os
+#import pandas as pd
+#import feather
+
+## Initialize Earth Engine
+ee.Initialize()
+
+#Source necessary functions. We do this instead of 'import' because of EE quirk.
+exec(open('GEE_pull_functions.py').read())
+
+#water = ee.Image("JRC/GSW1_1/GlobalSurfaceWater").select('occurrence').gt(80)
+
+## Bring in EE Assets
+# Deepest point for CONUS PGDL Lakes
+# DP Code available at https://zenodo.org/record/4136755#.X5d54pNKgUE
+dp = (ee.FeatureCollection('users/sntopp/USGS/PGDL_lakes_deepest_point')
+  .filterMetadata('distance', "greater_than", 60))
+
+#CONUS Boundary
+us = ee.FeatureCollection("USDOS/LSIB_SIMPLE/2017")\
+    .filterMetadata('country_na', 'equals', 'United States')
+
+## WRS Tiles in descending (daytime) mode for CONUS
+wrs = ee.FeatureCollection('users/sntopp/wrs2_asc_desc')\
+    .filterBounds(us)\
+    .filterMetadata('MODE', 'equals', 'D')
+    
+pr = wrs.aggregate_array('PR').getInfo()
+
+#Bring in temp data
+l8 = ee.ImageCollection("LANDSAT/LC08/C02/T1_L2")
+l7 = ee.ImageCollection("LANDSAT/LE07/C02/T1_L2")
+era5 = ee.ImageCollection("ECMWF/ERA5_LAND/HOURLY")
+
+#Standardize band names between the various collections and aggregate 
+#them into one image collection
+bn8 = ['ST_B10', 'ST_QA', 'QA_PIXEL']
+bn57 = ['ST_B6', 'ST_QA', 'QA_PIXEL']
+bns = ['temp','temp_qa','pixel_qa']
+  
+ls7 = l7.select(bn57, bns)
+ls8 = l8.select(bn8, bns)
+
+##Do coarse cloud filtering
+ls = ee.ImageCollection(ls7.merge(ls8))\
+    .filter(ee.Filter.lt('CLOUD_COVER', 50))\
```

Comment:
Curious and ignorant here: do you eliminate many clear-sky pixels by cutting out whole scenes with 50% clouds? And is that actually what you're doing here, or does this Filter operate at the pixel level?

Reply:
You're reading this correctly.  This is at the scene and not pixel level.  The cloud detection algorithms, especially pre-landsat 8 are mediocre, and generally when you have more than 50% clouds you also have a lot of undetected cirrus clouds.  From what I've read, clouds really mess up the emissivity calculations for surface temperature, so this initial filter is just a qa filter to account for undetected cirrus clouds and minimize the total number of scenes we need to process.

Reply:
We do pixel level cloud detection further down.